addons/point_of_sale/static/tests/unit/related_models_test.js

```diff
@@ -0,0 +1,948 @@
+                c1["<-product.product.category_id"],
+                [p1],
+                "Product should be linked to the category"
+            );
+            models["product.product"].update(p1, { category_id: [["clear"]] });
+            assert.notOk(
+                [c1["<-product.product.category_id"]].includes(p1),
+                "All categories should be unlinked from the product"
+            );
+        });
+
+        QUnit.test("delete operation, one2many item", (assert) => {
+            const p1 = models["product.product"].create({});
+            const p2 = models["product.product"].create({});
+            const o1 = models["pos.order"].create({});
+            o1.update({ product_ids: [["link", p1, p2]] });
+
+            assert.ok(o1.product_ids.includes(p1), "Product should be linked to the category");
+
+            p1.delete();
+            assert.notOk(models["product.product"].read(p1.id), "Product should be deleted");
+            assert.notOk(
+                o1.product_ids.includes(p1),
+                "Product should be unlinked from the category"
+            );
+        });
+
+        QUnit.test("delete operation, many2one item", (assert) => {
+            const p1 = models["product.product"].create({});
+            const c1 = models["product.category"].create({});
+            p1.update({ category_id: c1 });
+
+            assert.deepEqual(
+                c1["<-product.product.category_id"],
+                [p1],
+                "Product should be linked to the category"
+            );
+
+            c1.delete();
+            assert.notOk(models["product.category"].read(c1.id), "Category should be deleted");
+            assert.equal(p1.category_id, undefined, "Product should be unlinked from the category");
+        });
+    });
+
+    QUnit.module("many2many relations", (hooks) => {
+        let models;
+        hooks.beforeEach(() => {
+            [models] = createRelatedModels({
+                "product.product": {
+                    tag_ids: {
+                        type: "many2many",
+                        relation: "product.tag",
+                        relation_table: "product_tag_product_product_rel",
+                    },
+                },
+                "product.tag": {},
+            });
+        });
+
+        QUnit.test("create operation, create", (assert) => {
+            const tag1 = { name: "Electronics" };
+            const tag2 = { name: "New" };
+            const product = models["product.product"].create({
+                name: "Smartphone",
+                tag_ids: [["create", tag1, tag2]],
+            });
+            const readT1 = models["product.tag"].read(tag1.id);
+            assert.ok(product.tag_ids.includes(readT1), "tag1 should be linked with product");
+            assert.ok(
+                readT1["<-product.product.tag_ids"].includes(product),
+                "product should be backlinked to tag1"
+            );
+        });
```

Comment:
```suggestion
```

remove this. we should not rely on the id that is magically added by the system. It's an implementation detail.